infrastructure/aws/cdk/app.py

```diff
@@ -181,140 +91,120 @@ def __init__(  # noqa: C901
             vpc.add_gateway_endpoint(key, service=service)
 
         eodb_settings = eoDBSettings()
-        db = rds.DatabaseInstance(
+        
+        pgstac_db = PgStacDatabase(
             self,
-            f"{id}-postgres-db",
+            "pgstac-db",
             vpc=vpc,
-            engine=rds.DatabaseInstanceEngine.POSTGRES,
+            engine=rds.DatabaseInstanceEngine.postgres(
+                version=rds.PostgresEngineVersion.VER_14
+            ),
+            vpc_subnets=ec2.SubnetSelection(
+                subnet_type=ec2.SubnetType.PUBLIC
+            ),
+            allocated_storage=eodb_settings.allocated_storage,
             instance_type=ec2.InstanceType.of(
                 ec2.InstanceClass.BURSTABLE3,
                 ec2.InstanceSize(eodb_settings.instance_size),
             ),
             database_name="postgres",
-            # should set the subnet to `PRIVATE_ISOLATED` but then we need either a bastion host to connect to the db
-            # or an API to ingest/delete data in the DB
-            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
             backup_retention=Duration.days(7),
             deletion_protection=eoapi_settings.stage.lower() == "production",
             removal_policy=RemovalPolicy.SNAPSHOT
             if eoapi_settings.stage.lower() == "production"
             else RemovalPolicy.DESTROY,
-        )
-
-        setup_db = BootstrappedDb(
-            self,
-            "STAC DB for eoapi",
-            db=db,
-            new_dbname=eodb_settings.dbname,
-            new_username=eodb_settings.user,
-            secrets_prefix=os.path.join(stage, name),
-            pgstac_version=eodb_settings.pgstac_version,
-            enable_context=eodb_settings.context,
-            enable_mosaic_index=eodb_settings.mosaic_index,
-            context_dir=context_dir,
-        )
+            bootstrapper_props=BootstrapPgStacProps(custom_resource_properties={"pgstac_version": eodb_settings.pgstac_version,
+                "context": eodb_settings.context,
+                "mosaic_index": eodb_settings.mosaic_index},
+                lambda_function_options={},
+                pgstac_db_name=eodb_settings.dbname,
+                pgstac_username=eodb_settings.user,
+                secrets_prefix=os.path.join(stage, name)                                                                            
+        ))
 
         CfnOutput(
             self,
             f"{id}-database-secret-arn",
-            value=db.secret.secret_arn,
+            value=pgstac_db.pgstac_secret.secret_arn,
             description="Arn of the SecretsManager instance holding the connection info for Postgres DB",
         )
 
         # eoapi.raster
         if "raster" in eoapi_settings.functions:
+            
             db_secrets = {
-                "POSTGRES_HOST": setup_db.secret.secret_value_from_json(
+                "POSTGRES_HOST": pgstac_db.pgstac_secret.secret_value_from_json(
```

Comment:
we should be able to do away entirely with these because the secret arn is passed as an env var to the lambda but I'd have to change the runtime to parse the secret there, and I want to avoid changing the runtime at all in this repo, so I left it as is.

same comment applies for all lambdas where we pass the postgres env vars.